src/chia_dialect.rs

```diff
@@ -103,18 +107,26 @@ impl Dialect for ChiaDialect {
             34 => op_all,
             // 35 ---
             // 36 = softfork
-            _ => match extension {
-                OperatorSet::BLS => match b[0] {
-                    48 => op_coinid,
-                    // TODO: add BLS operators here
-                    _ => {
-                        return unknown_operator(allocator, o, argument_list, self.flags, max_cost);
+            _ => {
+                if extension == OperatorSet::BLS || (self.flags & ENABLE_BLS_OPS_OUTSIDE_GUARD) != 0
```

Comment:
I do worry a bit about complex logic like this sprouting in the dispatcher because a small mistake can result in historical behaviour changing. This might be why I'm so biased towards just having multiple dispatchers and choosing the correct one for the given dialect. There is a chance this one will end up not being permanent though, so it's probably good for now.

Reply:
I was thinking of different ways to express this to be as simple and robust as possible. I couldn't find a clear-cut improvement. For example, one could imagine this flag just controls what the initial state of `extension` is, so, instead of being `None` it's `OperatorSet::BLS`. This would make all programs be run as if they were inside the BLS softfork. However, it doesn't really model that the default operator set in the future could be a combination of other sets.

Although, now that I say it out loud; we won't have another hard fork, so that's probably not a reasonable case to consider. I'll try to make this a bit better in a future patch